### ISSUE TITLE
Rocket/nightly?

### ISSUE THREAD

**Issue by apendleton** [mapbox/Hecate#12]
Current Hecate relies on the [Rocket](https://github.com/SergioBenitez/Rocket) web framework, which requires the use of the unstable/nightly Rust compiler. Nightly contains language features that are subject to change, and also [breaks occasionally](https://github.com/rust-lang/rust/issues/40106), and at least per the current instructions (which recommend installing the latest nightly), each developer contributing to the project might end up with a different compiler with different features, behavior, and bugs depending on when they happen to set up their dev environment.

Rocket itself [doesn't seem particularly close](https://github.com/SergioBenitez/Rocket/issues/19) to running on stable (it seems like a project mostly interesting in pushing the language envelope rather than running on stable), and independent of stable vs unstable, the lead dev anticipates a [major API overhaul](https://github.com/SergioBenitez/Rocket#future-improvements) to move towards async as the Rust async story matures.

The ergonomics do seem nice, but in the interests of building on a solid foundation and making it easy for everyone to work on it, I wonder if we shouldn't consider moving to a framework with a stable API that runs on the stable compiler? [Nickel](http://nickel-org.github.io/) seems like a possibility, or if we wanted to get a head-start on async, maybe [Gotham](https://gotham.rs/) ?

**Comment by ingalls** [comment]
Per voice, the server was originally written with Iron as the backend. 

Due to problems with the underlying Iron codebase itself - around it handles threads and integrated with Hyper, I moved to Nickel and then settled on Rocket. This was a good move in hindsight as Iron is no longer maintained as of 3 days ago: https://github.com/iron/iron/pull/576#issue-167880860

Talking to the rocket user group on IRC leads me to believe they are being fairly careful about major breaking changes where possible to the external API but as you mentioned above do plan to break once Hyper/Tokio threads stabilize.

I only looked briefly at Gotham but shied away from it as it is still a 0.1 release and had very poor DB connection pooling at the time.

I would propose we stay with rocket with a hard version lock unless we run into a concrete problem.
